maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/Utf8RecodingDeferredFileOutputStream.java

```diff
@@ -93,14 +94,14 @@ public synchronized void write( String output, boolean newLine )
         }
         else
         {
-            cache.flip();
+            ( (Buffer) cache ).flip();
```

Comment:
Can we declare the `cache` as `Buffer` here as well?

Reply:
Again, at some point `array()` is called, returning `Object` instead of `byte[]`. Furthermore, `put()` is called twice, a method unavailable in `Buffer`.